### PR TITLE
Update CRD version to v1alpha2

### PR DESCRIPTION
**Pull request by JakeSCahill**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2725
Review deadline: 18 September

Note: I'm excluding the User types from the docs until the new User resource/controller is released.

https://github.com/redpanda-data/docs/blob/832e2ebc1c192a2db4e97f7fa13f61dda72207dc/.github/crd-config/config.yaml#L3-L22

## Page previews

[Preview](https://deploy-preview-768--redpanda-docs-preview.netlify.app/current/reference/k-crd/)

## Checks

- [ ] New feature
- [ ] Content gap
- [x] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: .github/crd-config/config.yaml [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+    - "UserTemplateSpec$"
+    - "Password$"
+    - "PasswordSource$"
+    - "ACLRule$"
+    - "ACLType$"
+    - "ACLOperation$"
+    - "PatternType$"
+    - "ResourceType$"
+    - "ACLResourceSpec$"
+    - "ClusterSource$"
```
Comment (andrewstucki) [inline .github/crd-config/config.yaml:19]:
```suggestion
    - "ClusterSource$"
    - "AdminAPISpec$"
    - "AdminSASL$"
    - "MetadataTemplate$"
```
A few more only used in User stuff currently.

**Review by andrewstucki — COMMENTED** [review]
Found a couple more we may want to ignore.

File: .github/crd-config/templates/asciidoctor/operator/gv_list.tpl [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
-.Packages
-{{- range $groupVersions }}
-- {{ asciidocRenderGVLink . }}
+// Autogenerated documentation. Do not edit this page See https://github.com/redpanda-data/docs/blob/main/.github/workflows/generate-crd.yml.
+{{- with index $groupVersions 0 }}
+= {{ . }}
 {{- end }}
+:anchor_prefix: k8s-api
+:description: Custom resource definitions for Redpanda resources. Use the Redpanda resources to create and manage Redpanda clusters, users and topics with the Redpanda Operator.
+:page-aliases: reference:k-topic-crd.adoc
```
Comment (RafalKorepta) [inline .github/crd-config/templates/asciidoctor/operator/gv_list.tpl:10]: I'm not sure if k-topic-crd.adoc is removed in this PR?

File: modules/manage/pages/kubernetes/k-manage-topics.adoc [hunk 3]
```diff
@@ -565,7 +565,7 @@ include::develop:partial$topic-defaults.adoc[]
 
 == Suggested reading
 
-- xref:reference:k-topic-crd.adoc[]
+- xref:reference:k--crd.adoc[]
```
Comment (RafalKorepta) [inline modules/manage/pages/kubernetes/k-manage-topics.adoc]: Should this be `k--crd.adoc` or `k-crd.adoc`? Single `-` or double?

Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/k-manage-topics.adoc]:
```suggestion
- xref:reference:k-crd.adoc[]
```

Comment (JakeSCahill) [reply modules/manage/pages/kubernetes/k-manage-topics.adoc]: thanks!

File: .github/crd-config/config.yaml (hunk 1 above)
Comment (JakeSCahill) [reply .github/crd-config/config.yaml:19]: thank you!

File: local-antora-playbook.yml [hunk 4]
```diff
@@ -10,6 +10,7 @@ urls:
   latest_version_segment: 'current'
 output:
   clean: true
+  dir: ./docs
```
Comment (JakeSCahill) [inline local-antora-playbook.yml]:
```suggestion
```

File: package.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
   "description": "Redpanda documentation",
   "license": "ISC",
   "scripts": {
-    "build": "antora --to-dir docs --fetch local-antora-playbook.yml",
+    "build": "antora --fetch local-antora-playbook.yml",
```
Comment (JakeSCahill) [inline package.json]:
```suggestion
    "build": "antora --to-dir docs --fetch local-antora-playbook.yml",
```

File: .github/crd-config/templates/asciidoctor/operator/gv_list.tpl (hunk 2 above)
Comment (JakeSCahill) [reply .github/crd-config/templates/asciidoctor/operator/gv_list.tpl:10]: yes we removed it in this PR.

Comment (JakeSCahill) [reply .github/crd-config/templates/asciidoctor/operator/gv_list.tpl:10]: this line adds the redirect from that old page to this one.

**Review by RafalKorepta — APPROVED** [review]

**Review by chrisseto — APPROVED** [review]

**Review by Feediver1 — APPROVED** [review]
lgtm

**Pull request merged**